### PR TITLE
Bug fix and pool manger tests

### PR DESCRIPTION
**Pull request by dguerin**
* Fixing some bugs in the pool manger flow
* Added some actor tests
* Changed event name

### CODE REVIEW

File: src/CaptainHook.PoolManagerActor/PoolManagerActor.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ namespace CaptainHook.PoolManagerActor
     public class PoolManagerActor : Actor, IPoolManagerActor
     {
         private readonly IBigBrother _bigBrother;
+        private readonly IActorProxyFactory _mockActorProxyFactory;
```
Comment (arturek) [inline src/CaptainHook.PoolManagerActor/PoolManagerActor.cs]: why 'mock...'? It's a standard actor proxy factory. It's mocked only when a test is executed.

File: src/CaptainHook.PoolManagerActor/PoolManagerActor.cs [hunk 2]
```diff
@@ -64,14 +67,20 @@ protected override async Task OnActivateAsync()
                 _free = new HashSet<int>(NumberOfHandlers);
                 _busy = new Dictionary<Guid, MessageHook>(NumberOfHandlers);
 
-                for (var i = 0; i < NumberOfHandlers; i++)
+                for (var i = 1; i < NumberOfHandlers + 1; i++)
```
Comment (arturek) [inline src/CaptainHook.PoolManagerActor/PoolManagerActor.cs]: maybe `Enumerable.Range(1, 5).ToHashSet()`

File: src/CaptainHook.PoolManagerActor/PoolManagerActor.cs (hunk 1 above)
Comment (dguerin) [reply src/CaptainHook.PoolManagerActor/PoolManagerActor.cs]: let me check, had it there for another reason but might be able to remove it.

File: src/CaptainHook.PoolManagerActor/PoolManagerActor.cs (hunk 2 above)
Comment (dguerin) [reply src/CaptainHook.PoolManagerActor/PoolManagerActor.cs]: lol.

**Pull request merged**